packages/mermaid-layout-elk/src/render.ts

```diff
@@ -24,27 +39,37 @@ export const render = async (
   const nodeDb: Record<string, any> = {};
   const clusterDb: Record<string, any> = {};
 
-  const addVertex = async (nodeEl: any, graph: { children: any[] }, nodeArr: any, node: any) => {
-    const labelData: any = { width: 0, height: 0 };
+  const addVertex = async (
+    nodeEl: SVGGroup,
+    graph: { children: NodeWithVertex[] },
+    nodeArr: Node[],
+    node: Node
+  ) => {
+    const labelData: LabelData = { width: 0, height: 0 };
 
-    let boundingBox;
-    const child = {
-      ...node,
-    };
-    graph.children.push(child);
-    nodeDb[node.id] = child;
     const config = getConfig();
 
     // Add the element to the DOM
     if (!node.isGroup) {
+      const child: NodeWithVertex = {
+        ...node,
+      };
+      graph.children.push(child);
+      nodeDb[node.id] = child;
+
       const childNodeEl = await insertNode(nodeEl, node, { config, dir: node.dir });
-      boundingBox = childNodeEl.node().getBBox();
+      const boundingBox = childNodeEl.node()!.getBBox();
```

Comment:
I agree.

I think in 95% of cases, `d3Selection.node()` will never return `null`, though, since we've created a node, so we know it exists. E.g. `elem.insert<SVGAElement>('svg:a').node()` will never return `null`, since we're creating a new `SVGALement`.

Unfortunately, I don't know of anyway to fix this without putting `!` everywhere.

Maybe we can adjust the official types of `d3-selection` https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6725a0f4009013cb59f7f57971568cdc287745a4/types/d3-selection/index.d.ts to have a new generic field like `NotEmpty = true`, that's automatically set when you create a D3.Selection, but I'd imagine that might be quite a bit of work too!